From fb5fa9f5e93eba61e12d4d66ad18e7f61bfdc677 Mon Sep 17 00:00:00 2001
From: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>
Date: Mon, 24 Feb 2025 13:21:11 +0100
Subject: [PATCH] native/posix arch: Add option to enable building of nsi_errno
 component

Add a new kconfig option to enable building the new component provided
in the native simulator, nsi_errno, which allows converting the errno
value between libCs.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>
---
 arch/posix/CMakeLists.txt          |  1 +
 arch/posix/Kconfig                 |  2 ++
 arch/posix/Kconfig.natsim_optional |  8 ++++++++
 arch/posix/natsim_optional.cmake   | 10 ++++++++++
 4 files changed, 21 insertions(+)
 create mode 100644 arch/posix/Kconfig.natsim_optional
 create mode 100644 arch/posix/natsim_optional.cmake

diff --git a/arch/posix/CMakeLists.txt b/arch/posix/CMakeLists.txt
index 8dfc696bc6d..3d7280daa52 100644
--- a/arch/posix/CMakeLists.txt
+++ b/arch/posix/CMakeLists.txt
@@ -199,4 +199,5 @@ if(NOT ${LLVM_SANITIZERS_ARG} STREQUAL "")
   target_compile_options(native_simulator INTERFACE ${LLVM_SANITIZERS_ARG})
 endif()
 
+include(natsim_optional.cmake)
 add_subdirectory(core)
diff --git a/arch/posix/Kconfig b/arch/posix/Kconfig
index 1ed79b38976..d67486288a5 100644
--- a/arch/posix/Kconfig
+++ b/arch/posix/Kconfig
@@ -41,4 +41,6 @@ config ARCH_POSIX_TRAP_ON_FATAL
 	  Raise a SIGTRAP signal on fatal error before exiting.
 	  This automatically suspends the target if a debugger is attached.
 
+rsource "Kconfig.natsim_optional"
+
 endmenu
diff --git a/arch/posix/Kconfig.natsim_optional b/arch/posix/Kconfig.natsim_optional
new file mode 100644
index 00000000000..4ec8dcdc4a5
--- /dev/null
+++ b/arch/posix/Kconfig.natsim_optional
@@ -0,0 +1,8 @@
+# Copyright (c) 2025 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+config NATIVE_USE_NSI_ERRNO
+	bool
+	help
+	  Build the native simulator nsi_errno component with the Zephyr embedded code.
+	  This component allows translating errno values from/to the embedded libC to/from the host libC.
diff --git a/arch/posix/natsim_optional.cmake b/arch/posix/natsim_optional.cmake
new file mode 100644
index 00000000000..e2f3d5f0eb3
--- /dev/null
+++ b/arch/posix/natsim_optional.cmake
@@ -0,0 +1,10 @@
+# Copyright (c) 2025 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+# Selection of optional components provided by the native simulator
+
+if(CONFIG_NATIVE_USE_NSI_ERRNO)
+  zephyr_library()
+
+  zephyr_library_sources(${NSI_DIR}/common/src/nsi_errno.c)
+endif()
-- 
2.48.1

